### PR TITLE
Paintable cleanup

### PR DESCRIPTION
**Pull request by JoergAtGithub**
Removed 2 unused methods
Changed code to return early style
Implemented missing SVG branch for Paintable::toImage()
Replaced hardcoded transparency color by Qt::transparent
Use std::move instead reset/release

### CODE REVIEW

File: src/widget/paintable.cpp [hunk 1]
```diff
@@ -103,60 +104,74 @@ bool Paintable::isNull() const {
...
-    //   return m_pPixmap->toImage();
-    // relying on QPixmap returning QImage() when it was null.
-    return !m_pPixmap ? QImage() : m_pPixmap->toImage();
+    if (m_pPixmap) {
+        return m_pPixmap->toImage();
+    }
+
+    if (m_pSvg) {
+        QImage image(m_pSvg->defaultSize(), QImage::Format_ARGB32);
+        image.fill(Qt::transparent);
```
Comment (daschuer) [inline src/widget/paintable.cpp:161]: Why is this required? Please add a source code comment.

**Review by daschuer — CHANGES_REQUESTED** [review]
LGTM, only a minor comment. 

Reading the code I notice that we render the SVG over and over again when for example called here:

https://github.com/mixxxdj/mixxx/blob/c0b4f50082b27767305357551c4c8f18b4db95cc/src/widget/wbattery.cpp#L168

Does it make sense to keep a copy of the QImage, and skip all the logic in draw, if the size did not change? I am not sure if this works, because I don't know when the SVG is exactly rendered and if there is already a caching involved.

File: src/widget/paintable.cpp [hunk 2]
```diff
@@ -103,60 +104,74 @@ bool Paintable::isNull() const {
...
-    // This confusion let to the wrong assumption that we could simple
-    //   return m_pPixmap->toImage();
-    // relying on QPixmap returning QImage() when it was null.
-    return !m_pPixmap ? QImage() : m_pPixmap->toImage();
+    if (m_pPixmap) {
+        return m_pPixmap->toImage();
+    }
+
+    if (m_pSvg) {
+        QImage image(m_pSvg->defaultSize(), QImage::Format_ARGB32);
```
Comment (Swiftb0y) [inline src/widget/paintable.cpp:160]:
```suggestion
        QImage image(m_pSvg->defaultSize(), QImage::Format_ARGB32_Premultiplied);
```
image

**Comment by JoergAtGithub** [comment]
> Reading the code I notice that we render the SVG over and over again when for example called here:
> 
> https://github.com/mixxxdj/mixxx/blob/c0b4f50082b27767305357551c4c8f18b4db95cc/src/widget/wbattery.cpp#L168

Yes we do, this is addressed in #12904 for 2.4 and not in this PR.

**Review by Swiftb0y — APPROVED** [review]
LGTM. waiting for @daschuer

**Review by daschuer — APPROVED** [review]
Thank you

**Pull request merged**